flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -178,7 +178,9 @@ public boolean isCheckpointingEnabled() {
      * Gets the checkpointing mode (exactly-once vs. at-least-once).
      *
      * @return The checkpointing mode.
+     * @deprecated Use {@link #getCheckpointMode} instead.
      */
+    @Deprecated
     public CheckpointingMode getCheckpointingMode() {
         return configuration.get(ExecutionCheckpointingOptions.CHECKPOINTING_MODE);
```

Comment:
IIUC, we should replace all internal usages of original CheckpointMode and docs with new one except for the user-facing interface, right?
I'd suggest to transform to the new CheckpointMode for all original public interfaces, and use the new one for other inner places.

Reply:
> IIUC, we should replace all internal usages of original CheckpointMode and docs with new one except for the user-facing interface, right?

Yeah, and this will be addressed by another PR.

> I'd suggest to transform to the new CheckpointMode for all original public interfaces, and use the new one for other inner places.

Well, I suggest we make this easier since the public API and the option will be removed at the same time, no more complex conversion needed, WDYT?